misc/src/openalea/misc/headache.py

```diff
@@ -0,0 +1,105 @@
+    headerhandler.file.write(tab+'Copyright 1995- '+str(datetime.now().year)+' Inria, Cirad, Inra')
+    headerhandler.file.write('\n'+tab+'\n')
+    config['authors'] = [i for i in config['authors'].split(', ') if not i == '']
+    config['authors_email'] = [i for i in config['authors_email'].split(', ') if not i =='']
+    if not len(config['authors']) == len(config['authors_email']):
+      raise MetainfoError('`authors` and `authors_email` fields for `mod` parameter are not compatible')
+    if len(config['authors']) > 1:
+      headerhandler.file.write(tab+'File authors:\n'+tab*2+'* '+('\n'+tab*2+'* ').join([i+' <'+j+'>' if not j == '' else i for i, j in zip(config['authors'], config['authors_email'])]))
+    else:
+      headerhandler.file.write(tab+'File author: '+config['authors'].pop()+' <'+config['authors_email'].pop()+'>')
+    headerhandler.write('\n'+tab+'\n')
+    config['contributors'] = [i for i in config['contributors'].split(', ') if not i == '']
+    config['contributors_email'] =  [i for i in config['contributors_email'].split(', ') if not i =='']
+    if not len(config['contributors']) == len(config['contributors_email']):
+      raise MetainfoError('`contributors` and `contributors_email` fields for `mod` parameter are not compatible')
+    if len(config['contributors']) > 0:
+      if len(config['contributors']) > 1:
+        headerhandler.file.write(tab+'File contributors:\n'+tab*2+'* '+('\n'+tab*2+'* ').join([i+' <'+j+'>' if not j == '' else i for i, j in zip(config['contributors'], config['contributors_email'])]))
+      else:
+        headerhandler.file.write(tab+'File contributor: '+config['contributors'].pop())
+      headerhandler.write('\n'+tab+'\n')
+    headerhandler.file.write(tab+'Distributed under the '+config['license']+'Cecill-C License.\n')
+    headerhandler.file.write(tab+'See accompanying file LICENSE.txt')
+    if not config['license_url'] == '':
+      headerhandler.file.write(' or copy at\n'+tab*2+config['license_url'])
+    headerhandler.file.write('\n'+tab+'\n'+tab+'OpenAlea WebSite: ')
+    headerhandler.file.write(config['url'])
+  except:
+    headerhandler.close()
+    os.remove(headerhandler.name)
+    raise
+  else:
+    headerhandler.close()
+
+  confighandler = NamedTemporaryFile(delete=False)
+  confighandler.file.write('| \".*\\\\.h\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.hpp\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.c\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.cpp\" -> frame open:\"/*\" line:\"*\" close:\"*/\"')
+  confighandler.file.write('| \".*\\\\.py\" -> frame open:\"#\" line:\"#\" close:\"#\"')
+  confighandler.close()
+
+  os.system('headache -c '+confighandler.name+' -h '+headerhandler.name+' '+' '.join([file for file in (rootpath/'src').walkfiles() if any([file.fnmatch(pattern) for pattern in patterns])]))
```

Comment:
Python [os.system doc](https://docs.python.org/2/library/os.html#os.system) suggests to use subprocess.
See [Replacing Older Functions with the subprocess Module](https://docs.python.org/2/library/subprocess.html#replacing-os-system)